src/decorators/types.js

```diff
@@ -1,9 +1,15 @@
 // @flow
 
+export type Cache = {
+  fnName: string,
+  keySets: Array<Array<string>>,
+  memoized: Function,
+};
+
 export type Decorator<TType> = (
-  target: TType,
-  name: $Keys<TType>,
   descriptor: Descriptor,
```

Comment:
I have flow error in this line: object type This type is incompatible with the expected param type of some incompatible instantiation of `TType` .. I have no idea why.